qa/task-cancellation/src/test/java/org/elasticsearch/tasks/TaskCancellationIT.java

```diff
@@ -0,0 +1,180 @@
+import org.elasticsearch.test.rest.yaml.ObjectPath;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.not;
+
+public class TaskCancellationIT extends ESRestTestCase {
+    private RestHighLevelClient oldCluster;
+    private RestHighLevelClient newCluster;
+
+    private List<HttpHost> readHosts(String prop) {
+        final String address = System.getProperty(prop);
+        assertNotNull("[" + prop + "] is not configured", address);
+        List<HttpHost> hosts = parseHosts(address);
+        assertThat("[" + prop + "] is empty", hosts, not(empty()));
+        return hosts;
+    }
+
+    void configureRemoteCluster(RestHighLevelClient client, String remoteCluster, List<HttpHost> seedNodes) throws Exception {
+        final String seeds = seedNodes.stream().map(HttpHost::toHostString).collect(Collectors.joining(","));
+        logger.info("Configure remote cluster [{}] with seed [{}]", remoteCluster, seeds);
+        final Request request = new Request("PUT", "/_cluster/settings");
+        request.setJsonEntity("{\"persistent\": {\"cluster.remote." + remoteCluster + ".seeds\": \"" + seeds + "\"}}");
+        client.cluster()
+            .putSettings(
+                new ClusterUpdateSettingsRequest().persistentSettings(
+                    Settings.builder().put("cluster.remote." + remoteCluster + ".seeds", seeds)
+                ),
+                RequestOptions.DEFAULT
+            );
+        assertBusy(() -> {
+            final RemoteInfoResponse remoteInfo = client.cluster().remoteInfo(new RemoteInfoRequest(), RequestOptions.DEFAULT);
+            assertThat(remoteInfo.getInfos(), not(empty()));
+            for (RemoteConnectionInfo info : remoteInfo.getInfos()) {
+                assertThat(info.getClusterAlias(), equalTo(remoteCluster));
+                assertTrue(info.isConnected());
+            }
+        });
+    }
+
+    Set<String> getNodeIds(RestHighLevelClient client) throws IOException {
+        Response response = client.getLowLevelClient().performRequest(new Request("GET", "_nodes"));
+        ObjectPath objectPath = ObjectPath.createFromResponse(response);
+        final Map<String, Object> nodeMap = objectPath.evaluate("nodes");
+        return nodeMap.keySet();
+    }
+
+    @Before
+    public void initClientsAndClusters() throws Exception {
+        RestClientBuilder.RequestConfigCallback callback = config -> config.setSocketTimeout(120 * 1000);
```

Comment:
we could alternatively configure this explicitly on the request to make it more explicit that we expect some requests to be blocked for awhile

Reply:
++, addressed in https://github.com/elastic/elasticsearch/pull/59828/commits/42b9173b88b003d6cfbcb0dbc8ee4452071f4b18